### PR TITLE
Docs Architecture

### PR DESCRIPTION
**Pull request by jde**
## What does this PR do?

Adds 2 architecture docs:
- plugins/plugin api/core
- tags

## How do I test this PR?

Read them.

### CODE REVIEW

**Comment by jde** [comment]
I see a couple of typos. Will clean them up asap. Please review for content only until further notice.

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+* Do I need to extend the Plugin API to support my plugin?
+
+Often times all the functionality a plugin needs is in the Core, but the Plugin API doesn't expose it. In these cases, we seek to iteratively extend the Talk Plugin API. All Plugin API contributions from the community must begin by [filing an Issue](https://github.com/coralproject/talk/blob/master/CONTRIBUTING.md).
+
+Note: we are stabilizing the process by which new Plugin API bindings are created, agreed upon and ultimately made part of our Plugins Contract. If you are interested in this process, please reach out to us.
+
+### Does it need updates to the Plugin API _and_ Core?
+
+3) What, if any, changes need to be made to Core so that the API can be extended?
```
Comment (wyattjoh) [inline docs/architecture.md]: Why the `3)`?

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+wait(60000);
+```
+
+Now, ask these questions:
+
+### Is it a Plugin?
+
+Most work for Talk these days happens in the Plugin space. If the answers to any of these questions is Yes, then you're thinking of a Plugin.
+
+* Does Talk's exiting Plugin APIs support the thing you want to build?
```
Comment (wyattjoh) [inline docs/architecture.md]: `existing`

File: docs/architecture-tags.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+* `permissions` information about who can see and set the tag,
+* `models` which `ITEM_TYPES` this tag can be applied to, and
+
+Whenever a tag is 'handled' by the server, it references this definition to determine that tag's behavior.
+
+See [Plugin API Documentation](plugins-server.html#field-tags) for more information.
+
+### Creating a Tag Definition
+
+Tag Definitions must be created before tags can be used. This ensures that users cannot push arbitrary tags to hack the system. It also allows devs to specify the behavioral characteristics of the tag.
```
Comment (wyattjoh) [inline docs/architecture-tags.md]: Replace with:

> Tag Definitions must be created in order for the system to determine what tags are permitted and available on the server side.

Comment (wyattjoh) [reply docs/architecture-tags.md]: Note as well that the tag definition does not contain any logic around it as this document seems to indicate.

**Review by kgardnr — APPROVED** [review]

**Pull request merged**